SIL.WritingSystems.Tests/SldrTests.cs

```diff
@@ -274,7 +288,39 @@ public void GetLdmlFile_NotModified_DoesntDownloadNewFile()
 				.HasAtLeastOneMatchForXpath("/ldml/identity/special/sil:identity[@source='cldr']", environment.NamespaceManager);
 		}
 
-		#region internal methods
+		[Test]
+		[Category("LongRunning")] // ~2 minutes
+		public void GetLdmlFile_GetsValidLDML([Values("false", "true")] string isStaging,
+			[Values("ar", "az", "bn", "de", "en", "es", "en-GB", "fa", "fr", "hi", "hu", "id", "km", "ko", "ml", "ms", "my",
+				"ne", "pt", "ru", "rw", "sw", "ta", "te", "th", "tr", "ur", "vi", "zh", "zh-CN")] string ietfLanguageTag)
+		{
+			DownloadAndVerifyLDML(isStaging, ietfLanguageTag);
+		}
+
+		[Test]
+		public void GetLdmlFile_GetsValidLDML([Values("ar", "en", "fr", "ko", "sw")] string ietfLanguageTag)
+		{
+			DownloadAndVerifyLDML("false", ietfLanguageTag);
+		}
+
+		private static void DownloadAndVerifyLDML(string isStaging, string ietfLanguageTag)
+		{
+			using var environment = new TestEnvironment(false);
+			var originalStagingValue = Environment.GetEnvironmentVariable(Sldr.SldrStaging);
+			try
+			{
+				Environment.SetEnvironmentVariable(Sldr.SldrStaging, isStaging);
+				var file = Path.Combine(environment.FilePath, environment.TryGetLdmlFile(ietfLanguageTag));
+				var adaptor = new LdmlDataMapper(new TestWritingSystemFactory());
+				adaptor.Read(file, new WritingSystemDefinition("en", "Latn", "US", string.Empty, "eng", false), ex => Assert.Fail(ex.ToString()));
```

Comment:
Hmm, I don't see much verification happening - unless that's happening in the LdmlDataMapper.

At a minimum I'd expect that we assert that `adaptor.Read` returns something.